src/main/java/com/google/devtools/build/lib/remote/RemoteActionFileSystem.java

```diff
@@ -370,10 +415,22 @@ private RemoteFileArtifactValue getRemoteInputMetadata(PathFragment path) {
     if (m != null && m.isRemote()) {
       return (RemoteFileArtifactValue) m;
     }
-
     return injectedMetadata.get(execPath);
   }
 
+  @Nullable
+  private TreeArtifactValue getRemoteInputTreeMetadata(PathFragment path) {
+    if (!path.startsWith(outputBase)) {
+      return null;
+    }
+    PathFragment execPath = path.relativeTo(execRoot);
+    TreeArtifactValue m = inputArtifactData.getTreeMetadata(execPath);
+    if (m != null && m.isEntirelyRemote()) {
```

Comment:
Do we have the case where the tree is not entirely remote? If so, how do we handle that?

Reply:
No, I don't believe this could ever occur, since two actions cannot produce the same tree artifact. So I think `!isEntirelyRemote` effectively means "is entirely local".

Reply:
True.